### ISSUE TITLE
Error generated by alt text containing ^

### ISSUE THREAD

**Issue by e-duckworth** [brucemiller/LaTeXML#2873]
latexml currently exits with an error `Error:unexpected:^ Script ^ can only appear in math mode` from the following code `\includegraphics[alt = {this is not x^2}]{foo}`.

I know that some people argue against using syntax like this in alt text but I don't think it should be up to latexml to flag this as an error.

**Comment by dginev** [comment]
>  I don't think it should be up to latexml to flag this as an error.

I would be fine if this was an error, or if this wasn't an error, but it definitely feels like it should be up to latexml to inform the user of how its internals were affected by unexpected content.

Why not wrap this in `$x^2$` or change it to `x\textsuperscript{2}` ?

**Comment by xworld21** [comment]
This sounds like https://github.com/dginev/latexml-oxide/issues/723. If LaTeX is detokenizing the argument and depositing as a string into the PDF, then LaTeXML should do the same, instead of digesting the content. (One should look at the tagging code, though.)

**Comment by e-duckworth** [comment]
I think of alt text as basically being raw/verbatim text for a screen reader.  As far as I have been able to find out there is no universal/authoritative definition of what is best practice for alt-text, and therefore different users might do it differently.  Some people might put "x squared" some "x^2" and some "$x^2$".  Some screen readers would read the last as "dollar sign x squared dollar sign" and that doesn't seem ideal to me (I've tested this Adobe Acrobat in the tagged PDF).

I don't think I know the different warning/error levels with latexml.  Mostly when I've seen the red "Error ..." thing I've found that the output has failed pretty badly.  So for me, I'd rather have a warning.

**Comment by dginev** [comment]
> Mostly when I've seen the red "Error ..." thing I've found that the output has failed pretty badly. 

This is an excellent understanding, as a document that encounters this message is indeed likely to have its output damaged (there can be cascading effects). `^` use in text mode is meant to be an Error.

In our numerous team discussions with Bruce, we have had consensus that it would be nice to allow more advanced uses of TeX in the "alt" fields, from the perspective of LaTeXML being an enabling technology for accessibility. That means that we can not treat everything as a letter/other catcode, so that text and math can be interleaved in a clean way (and so that we emit MathML for math fragments).

In my opinion, when using LaTeXML, your best bet for high quality accessible readouts here is to encode `$x^2$` for your math fragment example. Emitting the MathML 4 Intent attributes is on the project roadmap.

**Comment by e-duckworth** [comment]
OK, I'll follow your suggestion! And my apologies if my original recommendation was too pushy, like I get to decide what latexml should do!

**Comment by saitanallensantiago26-beep** [comment]
> latexml currently exits with an error `Error:unexpected:^ Script ^ can only appear in math mode` from the following code `\includegraphics[alt = {this is not x^2}]{foo}`.
> 
> I know that some people argue against using syntax like this in alt text but I don't think it should be up to latexml to flag this as an error.

**Comment by saitanallensantiago26-beep** [comment]
> latexml currently exits with an error `Error:unexpected:^ Script ^ can only appear in math mode` from the following code `\includegraphics[alt = {this is not x^2}]{foo}`.
> 
> I know that some people argue against using syntax like this in alt text but I don't think it should be up to latexml to flag this as an error.

**Comment by xworld21** [comment]
> latexml currently exits with an error `Error:unexpected:^ Script ^ can only appear in math mode` from the following code `\includegraphics[alt = {this is not x^2}]{foo}`.
> I know that some people argue against using syntax like this in alt text but I don't think it should be up to latexml to flag this as an error.

I agree. I appreciate LaTeXML wanting to have a more flexible model for alt text (although: we already have the PDF model of actualtext/alt, the HTML/ARIA model of label/labelledby/description/describedby, the SVG model of title/desc; do we really need more?), but changing the semantics of an existing LaTeX construct goes a bit further than that.
